src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -20,19 +20,20 @@ public bool TryReadTo(out ReadOnlySpan<T> span, T delimiter, bool advancePastDel
         {
             ReadOnlySpan<T> remaining = UnreadSpan;
             int index = remaining.IndexOf(delimiter);
+
             if (index != -1)
             {
                 span = index == 0 ? default : remaining.Slice(0, index);
-                Advance(index + (advancePastDelimiter ? 1 : 0));
+                AdvanceCurrentSpan(index + (advancePastDelimiter ? 1 : 0));
```

Comment:
Does the `count < 0` check really make much of a difference?

Reply:
Not huge, no. But the code is smaller.